resqpy/multiprocessing/multiprocessing.py

```diff
@@ -0,0 +1,112 @@
+"""Multiprocessing module containing the function used to run the wrapperfunctions in parallel."""
+
+import logging
+from typing import List, Dict, Any, Callable, Union
+from pathlib import Path
+from resqpy.model import Model, new_model
+from joblib import Parallel, delayed, parallel_backend  # type: ignore
+
+log = logging.getLogger(__name__)
+
+
+def rm_tree(path: Union[Path, str]) -> None:
+    """Removes a directory using a pathlib Path.
+
+    Args:
+        path (Path/str): pathlib Path or string of the directory path.
+
+    Returns:
+        None.
+    """
+    path = Path(path)
+    for child in path.iterdir():
+        if child.is_file():
+            child.unlink()
+        else:
+            rm_tree(child)
+    path.rmdir()
+
+
+def function_multiprocessing(
+    function: Callable,
+    kwargs_list: List[Dict[str, Any]],
+    recombined_epc: Union[Path, str],
+    cluster,
+    consolidate: bool = True,
+) -> List[bool]:
+    """Calls a function concurrently with the specfied arguments.
+
+    A multiprocessing pool is used to call the function multiple times in parallel. Once
+    all results are returned, they are combined into a single epc file.
+
+    Args:
+        function (Callable): the function to be called. Needs to return:
+
+            - index (int): the index of the kwargs in the kwargs_list.
+            - success (bool): whether the function call was successful, whatever that
+                definiton is.
+            - epc_file (Path/str): the epc file path where the objects are stored.
+            - uuid_list (List[str]): list of UUIDs of relevant objects.
+
+        kwargs_list (List[Dict[Any]]): A list of keyword argument dictionaries that are
+            used when calling the function.
+        recombined_epc (Path/str): A pathlib Path or path string of
+            where the combined epc will be saved.
+        cluster (LocalCluster/JobQueueCluster): a LocalCluster is a Dask cluster on a
+            local machine. If using a job queing system, a JobQueueCluster can be used
+            such as an SGECluster, SLURMCluster, PBSCluster, LSFCluster etc.
+        consolidate (bool): if True and an equivalent part already exists in
+            a model, it is not duplicated and the uuids are noted as equivalent.
+
+    Returns:
+        success_list (List[bool]): A boolean list of successful function calls.
+    """
```

Comment:
Could be good to add a note to the docstring, saying that dask is used as the backend and needs to have been installed in the python environment (for people who don't read the tutorial!)

Reply:
I can add the link to the tutorial once it is finished too